packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -120,3 +120,34 @@ export function generateTemplateMetadata(state: State, codeGen: CodeGen): t.Stat
 
     return metadataExpressions;
 }
+
+const DECLARATION_DELIMITER = /;(?![^(]*\))/g;
+const PROPERTY_DELIMITER = /:(.+)/;
+
+// Borrowed from Vue template compiler.
+// https://github.com/vuejs/vue/blob/531371b818b0e31a989a06df43789728f23dc4e8/src/platforms/web/util/style.js#L5-L16
+export function parseStyleText(cssText: string): { [name: string]: string } {
+    const styleMap: { [name: string]: string } = {};
+
+    const declarations = cssText.split(DECLARATION_DELIMITER);
+    for (const declaration of declarations) {
+        if (declaration) {
+            const [prop, value] = declaration.split(PROPERTY_DELIMITER);
+
+            if (prop !== undefined && value !== undefined) {
+                styleMap[prop.trim()] = value.trim();
+            }
+        }
+    }
+
+    return styleMap;
+}
```

Comment:
I'm surprised that a regex is good enough for parsing style text. But if Vue is using it, it must be reasonable.

I suppose if we switched to something like PostCSS, you'd be worried about a regression in the performance of the compiler?

Reply:
We have been rolling with this for years now. It is not perfect but it does the trick.
I don't see any issue with moving to PostCSS to parse the CSS properties, that said it might be little overkill.